File: lib/screens/coach/coach_panel_screen.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:url_launcher/url_launcher.dart';
import '../../utils/colors.dart';
import '../../models/user_model.dart';
import '../../services/auth_service.dart';
import '../../services/team_service.dart';
import '../../data/jugadores_data.dart';
import 'tactical_board_screen.dart';
import 'physical_training_screen.dart';

class CoachPanelScreen extends StatefulWidget {
  const CoachPanelScreen({super.key});

  @override
  State<CoachPanelScreen> createState() => _CoachPanelScreenState();
}

class _CoachPanelScreenState extends State<CoachPanelScreen> {
  final _authService = AuthService();
  final _teamService = TeamService();
  UserModel? _user;
  final List<String> _selectedPlayers = [];

  @override
  void initState() {
    super.initState();
    _loadUser();
  }

  Future<void> _loadUser() async {
    final user = await _authService.getCurrentUser();
    if (user != null) {
      final savedCallUp = await _teamService.getCallUpList(user.club.id, user.division ?? 'General');
      setState(() {
        _user = user;
        _selectedPlayers.addAll(savedCallUp);
      });
    }
  }

  void _openWhatsApp() async {
    const String whatsappLink = 'https://chat.whatsapp.com/invite/example';
    final Uri url = Uri.parse(whatsappLink);
    if (!await launchUrl(url, mode: LaunchMode.externalApplication)) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          const SnackBar(content: Text('No se pudo abrir el enlace de WhatsApp')),
        );
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    final bool isPrepFisico = _user?.rolCuerpoTecnico == 'Preparador Físico';

    return Scaffold(
      backgroundColor: AppColors.background,
      body: Column(
        children: [
          Container(
            padding: const EdgeInsets.fromLTRB(30, 60, 30, 40),
            decoration: const BoxDecoration(
              color: AppColors.primary,
              borderRadius: BorderRadius.only(bottomLeft: Radius.circular(32), bottomRight: Radius.circular(32)),
              boxShadow: [BoxShadow(color: Colors.black26, blurRadius: 15, offset: Offset(0, 5))],
            ),
            child: Row(
              children: [
                Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text('ESTRATEGIA', style: GoogleFonts.montserrat(color: AppColors.secondary, fontSize: 10, fontWeight: FontWeight.w900, letterSpacing: 2)),
                    const SizedBox(height: 4),
                    Text('PANEL TÉCNICO', style: GoogleFonts.montserrat(color: Colors.white, fontSize: 24, fontWeight: FontWeight.w900, letterSpacing: -1)),
                  ],
                ),
                const Spacer(),
                Container(
                  padding: const EdgeInsets.all(12),
                  decoration: BoxDecoration(color: Colors.white.withOpacity(0.1), shape: BoxShape.circle),
                  child: const Icon(Icons.admin_panel_settings_rounded, color: Colors.white, size: 30),
                ),
              ],
            ),
          ),

          Expanded(
            child: ListView(
              padding: const EdgeInsets.all(24),
              children: [
                _buildActionCard(
                  'PIZARRA TÁCTICA',
                  'Diseñá jugadas y posicionamientos.',
                  Icons.gesture_rounded,
                  AppColors.primary,
                  () => Navigator.push(context, MaterialPageRoute(builder: (_) => const TacticalBoardScreen())),
                ),
                _buildActionCard(
                  'PLANIFICACIÓN FÍSICA',
                  'Enviá indicaciones de trabajo al plantel.',
                  Icons.fitness_center_rounded,
                  isPrepFisico ? AppColors.secondary : AppColors.info,
                  () {
                    if (_user != null) {
                      Navigator.push(context, MaterialPageRoute(builder: (_) => PhysicalTrainingScreen(coach: _user!)));
                    }
                  },
                  tag: isPrepFisico ? 'TU ESPECIALIDAD' : null,
                ),
                _buildActionCard(
                  'GRUPO DE JUGADORES',
                  'Chat directo en WhatsApp con el equipo.',
                  Icons.chat_bubble_rounded,
                  Colors.green.shade600,
                  _openWhatsApp,
                ),
                _buildActionCard(
                  'CONVOCATORIA',
                  'Armá la lista para el próximo partido.',
                  Icons.assignment_ind_rounded,
                  AppColors.secondary,
                  () => _showCallUpSheet(),
                ),
                _buildActionCard(
                  'RED DE DIRECTORES',
                  'Contacto con otros cuerpos técnicos.',
                  Icons.connect_without_contact_rounded,
                  Colors.blue.shade700,
                  () => _showCoachesNetwork(),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildActionCard(String title, String subtitle, IconData icon, Color color, VoidCallback onTap, {String? tag}) {
    return Container(
      margin: const EdgeInsets.only(bottom: 20),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(24),
        boxShadow: [BoxShadow(color: Colors.black.withOpacity(0.03), blurRadius: 10, offset: const Offset(0, 4))],
      ),
      child: Stack(
        children: [
          ListTile(
            onTap: onTap,
            contentPadding: const EdgeInsets.all(20),
            leading: Container(
              width: 55, height: 55,
              decoration: BoxDecoration(color: color.withOpacity(0.1), borderRadius: BorderRadius.circular(16)),
              child: Icon(icon, color: color, size: 28),
            ),
            title: Text(title, style: GoogleFonts.montserrat(fontWeight: FontWeight.w900, fontSize: 13, letterSpacing: 0.5)),
            subtitle: Padding(
              padding: const EdgeInsets.only(top: 4),
              child: Text(subtitle, style: const TextStyle(fontSize: 11, color: Colors.grey, height: 1.3)),
            ),
            trailing: const Icon(Icons.arrow_forward_ios_rounded, color: Colors.grey, size: 14),
          ),
          if (tag != null)
            Positioned(
              top: 10, right: 20,
              child: Container(
                padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 2),
                decoration: BoxDecoration(color: color, borderRadius: BorderRadius.circular(8)),
                child: Text(tag, style: const TextStyle(color: Colors.white, fontSize: 7, fontWeight: FontWeight.bold)),
              ),
            ),
        ],
      ),
    );
  }

  void _showCallUpSheet() {
    if (_user == null) return;
    final players = JugadoresData.getJugadoresPorClub(_user!.club.id);

    showModalBottomSheet(
      context: context,
      isScrollControlled: true,
      backgroundColor: Colors.transparent,
      builder: (context) => StatefulBuilder(
        builder: (context, setModalState) => Container(
          height: MediaQuery.of(context).size.height * 0.85,
          decoration: const BoxDecoration(color: Colors.white, borderRadius: BorderRadius.vertical(top: Radius.circular(32))),
          child: Column(
            children: [
              const SizedBox(height: 12),
              Container(width: 40, height: 4, decoration: BoxDecoration(color: Colors.grey.shade300, borderRadius: BorderRadius.circular(2))),
              Padding(
                padding: const EdgeInsets.all(24),
                child: Row(
                  children: [
                    const Icon(Icons.assignment_ind_rounded, color: AppColors.primary),
                    const SizedBox(width: 12),
                    Text('CONVOCATORIA', style: GoogleFonts.montserrat(fontWeight: FontWeight.w900, fontSize: 18, letterSpacing: -0.5)),
                    const Spacer(),
                    Text('${_selectedPlayers.length} / 18', style: const TextStyle(fontWeight: FontWeight.bold, color: Colors.grey)),
                  ],
                ),
              ),
              const Divider(height: 1),
              Expanded(
                child: players.isEmpty 
                  ? const Center(child: Text('No hay jugadores registrados en tu club.'))
                  : ListView.builder(
                      padding: const EdgeInsets.symmetric(vertical: 10),
                      itemCount: players.length,
                      itemBuilder: (context, i) {
                        final p = players[i];
                        final isSel = _selectedPlayers.contains(p.id);
                        return CheckboxListTile(
                          activeColor: AppColors.primary,
                          value: isSel,
                          onChanged: (v) {
                            setModalState(() {
                              if (v == true) {
                                _selectedPlayers.add(p.id);
                              } else {
                                _selectedPlayers.remove(p.id);
                              }
                            });
                          },
                          title: Text(p.nombre, style: GoogleFonts.montserrat(fontWeight: FontWeight.bold, fontSize: 14)),
                          subtitle: Text('${p.posicion} • #${p.numeroCamiseta}', style: const TextStyle(fontSize: 12)),
                          secondary: CircleAvatar(
                            backgroundColor: AppColors.primary.withOpacity(0.1),
                            child: Text(p.nombre[0], style: const TextStyle(color: AppColors.primary, fontWeight: FontWeight.bold)),
                          ),
                        );
                      },
                    ),
              ),
              Padding(
                padding: const EdgeInsets.all(24),
                child: SizedBox(
                  width: double.infinity,
                  height: 55,
                  child: ElevatedButton(
                    onPressed: () async {
                      if (_user != null) {
                        final nav = Navigator.of(context);
                        final sm = ScaffoldMessenger.of(context);
                        await _teamService.saveCallUpList(_user!.club.id, _user!.division ?? 'General', _selectedPlayers);
                        nav.pop();
                        sm.showSnackBar(const SnackBar(content: Text('Citación guardada y enviada.'), backgroundColor: AppColors.success));
                      }
                    },
                    style: ElevatedButton.styleFrom(backgroundColor: AppColors.primary, foregroundColor: Colors.white),
                    child: Text('CONFIRMAR CITACIÓN', style: GoogleFonts.montserrat(fontWeight: FontWeight.bold, letterSpacing: 1)),
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  void _showCoachesNetwork() {
    final coaches = [
      {'name': 'Ricardo Gareca', 'club': 'Belgrano Athletic', 'role': 'Head Coach'},
      {'name': 'Marcelo Bielsa', 'club': 'Alumni', 'role': 'Director Técnico'},
      {'name': 'Lionel Scaloni', 'club': 'River Plate', 'role': 'Preparador Físico'},
      {'name': 'Pep Guardiola', 'club': 'CUBA', 'role': 'Asistente'},
    ];

    showModalBottomSheet(
      context: context,
      isScrollControlled: true,
      backgroundColor: Colors.transparent,
      builder: (context) => Container(
        height: MediaQuery.of(context).size.height * 0.7,
        decoration: const BoxDecoration(color: Colors.white, borderRadius: BorderRadius.vertical(top: Radius.circular(32))),
        child: Column(
          children: [
            const SizedBox(height: 12),
            Container(width: 40, height: 4, decoration: BoxDecoration(color: Colors.grey.shade300, borderRadius: BorderRadius.circular(2))),
            Padding(
              padding: const EdgeInsets.all(24),
              child: Row(
                children: [
                  const Icon(Icons.connect_without_contact_rounded, color: Colors.blue),
                  const SizedBox(width: 12),
                  Text('RED DE TÉCNICOS', style: GoogleFonts.montserrat(fontWeight: FontWeight.w900, fontSize: 18, letterSpacing: -0.5)),
                ],
              ),
            ),
            const Divider(height: 1),
            Expanded(
              child: ListView.builder(
                padding: const EdgeInsets.all(20),
                itemCount: coaches.length,
                itemBuilder: (context, i) {
                  final c = coaches[i];
                  return Container(
                    margin: const EdgeInsets.only(bottom: 12),
                    decoration: BoxDecoration(color: Colors.grey.shade50, borderRadius: BorderRadius.circular(16)),
                    child: ListTile(
                      leading: const CircleAvatar(backgroundColor: Colors.blue, child: Icon(Icons.person, color: Colors.white)),
                      title: Text(c['name']!, style: GoogleFonts.montserrat(fontWeight: FontWeight.bold, fontSize: 14)),
                      subtitle: Text('${c['role']} en ${c['club']}'),
                      trailing: IconButton(icon: const Icon(Icons.chat_outlined, color: Colors.blue), onPressed: () {}),
                    ),
                  );
                },
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: app/src/main/java/com/example/hockey_app/ui/screens/coach/CoachPanelScreen.kt
package com.example.hockey_app.ui.screens.coach

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowForwardIos
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun CoachPanelScreen(
    onMenuClick: () -> Unit,
    onNavigateToTacticalBoard: () -> Unit,
    onNavigateToCallUpManagement: () -> Unit,
    onNavigateToSearchPlayers: () -> Unit,
    onNavigateToPhysicalPlanning: () -> Unit,
    viewModel: CoachViewModel = hiltViewModel(),
) {
    val context = androidx.compose.ui.platform.LocalContext.current
    // UI logic here

    Scaffold(
        topBar = {
            TopAppBar(
                title = { },
                navigationIcon = {
                    IconButton(onClick = onMenuClick) {
                        Icon(Icons.Default.Menu, contentDescription = "Menu", tint = Color.White)
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = MaterialTheme.colorScheme.primary)
            )
        },
        containerColor = Color(0xFFF5F5F5)
    ) { padding ->
        Column(modifier = Modifier.fillMaxSize().padding(padding)) {
            // Header
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(bottomStart = 32.dp, bottomEnd = 32.dp))
                    .background(MaterialTheme.colorScheme.primary)
                    .padding(start = 30.dp, end = 30.dp, bottom = 40.dp, top = 0.dp)
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Column {
                        Text(
                            "ESTRATEGIA",
                            color = MaterialTheme.colorScheme.secondary,
                            fontSize = 10.sp,
                            fontWeight = FontWeight.Black,
                            letterSpacing = 2.sp
                        )
                        Spacer(modifier = Modifier.height(4.dp))
                        Text(
                            "PANEL TÉCNICO",
                            color = Color.White,
                            fontWeight = FontWeight.Black,
                            fontSize = 24.sp,
                            letterSpacing = (-1).sp
                        )
                    }
                    Spacer(modifier = Modifier.weight(1f))
                    Box(
                        modifier = Modifier
                            .size(55.dp)
                            .background(Color.White.copy(alpha = 0.1f), CircleShape),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(Icons.Default.AdminPanelSettings, contentDescription = null, tint = Color.White, modifier = Modifier.size(30.dp))
                    }
                }
            }

            LazyColumn(
                modifier = Modifier.fillMaxSize(),
                contentPadding = PaddingValues(24.dp),
                verticalArrangement = Arrangement.spacedBy(20.dp)
            ) {
                item {
                    ActionCard(
                        "PIZARRA TÁCTICA",
                        "Diseñá jugadas y posicionamientos.",
                        Icons.Default.Gesture,
                        MaterialTheme.colorScheme.primary,
                        onClick = onNavigateToTacticalBoard
                    )
                }
                item {
                    ActionCard(
                        "PLANIFICACIÓN FÍSICA",
                        "Enviá indicaciones de trabajo al plantel.",
                        Icons.Default.FitnessCenter,
                        Color(0xFF2196F3),
                        onClick = onNavigateToPhysicalPlanning
                    )
                }
                item {
                    ActionCard(
                        "GRUPO DE JUGADORES",
                        "Chat directo en WhatsApp con el equipo.",
                        Icons.Default.ChatBubble,
                        Color(0xFF4CAF50)
                    ) {
                        com.example.hockey_app.utils.IntentUtils.openWhatsApp(context)
                    }
                }
                item {
                    ActionCard(
                        "CONVOCATORIA",
                        "Armá la lista para el próximo partido.",
                        Icons.Default.AssignmentInd,
                        MaterialTheme.colorScheme.secondary,
                        onClick = onNavigateToCallUpManagement
                    )
                }
                item {
                    ActionCard(
                        "RED DE DIRECTORES",
                        "Contacto con otros cuerpos técnicos.",
                        Icons.Default.ConnectWithoutContact,
                        Color(0xFF1976D2),
                        onClick = onNavigateToSearchPlayers
                    )
                }
            }
        }
    }
}

@Composable
fun ActionCard(
    title: String,
    subtitle: String,
    icon: ImageVector,
    color: Color,
    onClick: () -> Unit
) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .shadow(10.dp, RoundedCornerShape(24.dp), ambientColor = Color.Black.copy(alpha = 0.03f)),
        shape = RoundedCornerShape(24.dp),
        colors = CardDefaults.cardColors(containerColor = Color.White),
        elevation = CardDefaults.cardElevation(0.dp)
    ) {
        ListItem(
            modifier = Modifier.padding(10.dp),
            leadingContent = {
                Box(
                    modifier = Modifier
                        .size(55.dp)
                        .background(color.copy(alpha = 0.1f), RoundedCornerShape(16.dp)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(icon, contentDescription = null, tint = color, modifier = Modifier.size(28.dp))
                }
            },
            headlineContent = {
                Text(title, fontWeight = FontWeight.Black, fontSize = 13.sp, letterSpacing = 0.5.sp)
            },
            supportingContent = {
                Text(subtitle, fontSize = 11.sp, color = Color.Gray, lineHeight = 16.sp)
            },
            trailingContent = {
                Icon(Icons.AutoMirrored.Filled.ArrowForwardIos, contentDescription = null, tint = Color.LightGray, modifier = Modifier.size(14.dp))
            },
            colors = ListItemDefaults.colors(containerColor = Color.Transparent)
        )
    }
}
